tests/test_bot.py

```diff
@@ -116,6 +116,22 @@ def test_detect_self_bump(self, _) -> None:
 
         self.cleanup()
 
+    def test_detect_url(self, _) -> None:
+        """Test the ability of KarmnaBot to detect a URL which contains either the ++ or
+        -- tokens.
+        """
+        cases = [
+            (True, {"user": "GraceHopper", "text": "https://www.example--.com"}),
+            (False, {"user": "GraceHopper", "text": "https://www.example.com"}),
+        ]
+        bot = KarmaBot(
```

Comment:
We can probably mock out the token here so that the Unit Tests do not need the token to be in the environment.